StaticAnalyzer/views/android.py

```diff
@@ -1055,7 +1079,186 @@ def Strings(APP_FILE, APP_DIR, TOOLS_DIR):
         return dat
     except:
         PrintException("[ERROR] Extracting Strings from APK")
-
+# Esteve 14.08.2016 - begin - Pirated and Malicious App Detection with APKiD 
+def APKiD(APP_FILE,APP_DIR,TOOLS_DIR,APP_NAME):
+    try:
+        print "[INFO] Detecting Packers, Obfuscators, Compilers, and other stuff with APKiD"
+        RET=''
+        apkid=TOOLS_DIR+'apkid'
+        args=[apkid,'-j',APP_DIR+APP_FILE]
+        dat=ast.literal_eval(subprocess.check_output(args))
+        for key1, value1 in dat.items():
+            if key1.find('!') == -1:
+                file=APP_NAME
+            else:
+                file=key1.split('!')[1]
+            for key2, value2 in value1.items():
+                concept=key2
+                for item3 in value2:
+                    item=item3
+                    detection = False
+                    if concept == 'compiler' and item == 'Android SDK (dx)':
+                        RET=RET +'<tr><td>File <strong>'+file+'</strong> has been compiled using the <strong>'+item3+'</strong> compiler.</td><td><span class="label label-info">info</span></td><td> The file has been compiled using the standard Android SDK compiler.</td></tr>'
+                        detection = True
+                    if concept == 'compiler' and item == 'Android SDK (dexmerge)':
+                        RET=RET +'<tr><td>File <strong>'+file+'</strong> has been compiled using the <strong>'+item3+'</strong> compiler.</td><td><span class="label label-info">info</span></td><td> The file has been compiled using dexmerge, which is used for incremental builds by some IDEs (after using dx).</td></tr>'                 
+                        detection = True
+                    if concept == 'compiler' and (item == 'dexlib 1.x' or item == 'dexlib 2.x' or item == 'dexlib 2.x beta'):
+                        RET=RET +'<tr><td>File <strong>'+file+'</strong> has been compiled using the <strong>'+item3+'</strong> compiler.</td><td><span class="label label-danger">high</span></td><td> The file has been compiled using one of the dexlib families. This is an indicator of potential crack or malware injection.</td></tr>'
```

Comment:
What made you conculde that dexlib is always used by crack/malware developers. Won't there be any legitimate users?

I guess we need to reworld this and change the threat level to warning. What do you say?